src/NServiceBus.Core/Pipeline/Incoming/IncomingLogicalMessageContext.cs

```diff
@@ -36,7 +36,7 @@ public void UpdateMessageInstance(object newInstance)
                 return;
             }
 
-            var factory = this.Builder.Build<LogicalMessageFactory>();
+            var factory = Builder.Build<LogicalMessageFactory>();
```

Comment:
I know this is just cleanup, but since it's unrelated, should it be done somewhere else?